editor/property_selector.cpp

```diff
@@ -338,46 +338,24 @@ void PropertySelector::_item_selected() {
 	}
 	String name = item->get_metadata(0);
 
-	String class_type;
-	if (type != Variant::NIL) {
-		class_type = Variant::get_type_name(type);
-
-	} else {
-		class_type = base_type;
-	}
-
 	DocData *dd = EditorHelp::get_doc_data();
 	String text;
 
 	if (properties) {
-		String at_class = class_type;
-
-		while (at_class != String()) {
-			Map<String, DocData::ClassDoc>::Element *E = dd->class_list.find(at_class);
-			if (E) {
-				for (int i = 0; i < E->get().properties.size(); i++) {
-					if (E->get().properties[i].name == name) {
-						text = DTR(E->get().properties[i].description);
-					}
+		for (Map<String, DocData::ClassDoc>::Element *E = dd->class_list.front(); E; E = E->next()) {
+			for (int i = 0; i < E->get().properties.size(); i++) {
+				if (E->get().properties[i].name == name) {
+					text = E->get().properties[i].description;
```

Comment:
Sorry for the late follow-up. This doesn't look correct to me (though I haven't tested): if I understand correctly, it's going through all registered classes and picking the first property it finds with that name. That's not guaranteed to be the actual property that we're looking at - different classes can have unrelated properties with the same name.

That's why the previous logic starts at the current class, then walks up through the inheritance tree to find the property if it's not in the current class, as it can come from an ancestor class.